test/conformance/test_suite_test.go

```diff
@@ -2,81 +2,55 @@ package conformance
 
 import (
 	"flag"
-	"fmt"
-	"os"
+	"log"
 	"path"
 	"testing"
-
-	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/clean"
+	"time"
 
 	. "github.com/onsi/ginkgo/v2"
-	"github.com/onsi/ginkgo/v2/reporters"
-	"github.com/onsi/ginkgo/v2/types"
 	. "github.com/onsi/gomega"
 
-	testclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	kniK8sReporter "github.com/openshift-kni/k8sreporter"
 
 	// Test files in this package must not end with `_test.go` suffix, as they are imported as go package
 	_ "github.com/k8snetworkplumbingwg/sriov-network-operator/test/conformance/tests"
 
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/clean"
 	"github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/k8sreporter"
 )
 
 var (
-	junitPath      *string
-	dumpOutput     *bool
-	reporterFile   string
-	customReporter *k8sreporter.KubernetesReporter
+	customReporter *kniK8sReporter.KubernetesReporter
+	err            error
+	reportPath     *string
 )
 
 func init() {
-	dumpOutput = flag.Bool("dump", false, "dump informations for failed tests")
-	junitPath = flag.String("junit", "", "the path for the junit format report")
+	reportPath = flag.String("report", "", "the path of the report directory containing details for failed tests")
 }
 
 func TestTest(t *testing.T) {
-	RegisterFailHandler(Fail)
-
-	reporterFile = os.Getenv("REPORTER_OUTPUT")
-
-	clients := testclient.New("")
+	// We want to collect logs before any resource is deleted in AfterEach, so we register the global fail handler
+	// in a way such that the reporter's Dump is always called before the default Fail.
+	RegisterFailHandler(
+		func(message string, callerSkip ...int) {
+			if customReporter != nil {
+				customReporter.Dump(10*time.Minute, CurrentSpecReport().FullText())
+			}
+			Fail(message, callerSkip...)
```

Comment:
Please consider increasing `callerSkip` slice, as described here:

- https://github.com/openshift-kni/cnf-features-deploy/pull/1644

```suggestion
                        // Ensure failing line location is not affected by this wrapper
			for i := range callerSkip {
				callerSkip[i]++
			}
			Fail(message, callerSkip...)
```

Reply:
Done. thanks!